homeassistant/components/overkiz/climate_entities/hitachi_air_to_air_heat_pump.py

```diff
@@ -0,0 +1,359 @@
+
+HVAC_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_HVAC_MODES.items()}
+
+OVERKIZ_TO_SWING_MODES: dict[str, str] = {
+    OverkizCommandParam.BOTH: SWING_BOTH,
+    OverkizCommandParam.HORIZONTAL: SWING_HORIZONTAL,
+    OverkizCommandParam.STOP: SWING_OFF,
+    OverkizCommandParam.VERTICAL: SWING_VERTICAL,
+}
+
+SWING_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_SWING_MODES.items()}
+
+# Fan modes differ between OVP and HLRRWIFI protocol
+OVERKIZ_TO_FAN_MODES: dict[Protocol, dict[str, str]] = {
+    Protocol.OVP: {
+        OverkizCommandParam.AUTO: FAN_AUTO,
+        OverkizCommandParam.HIGH: FAN_HIGH,  # fallback, state can be exposed as MEDIUM, new state = med
+        OverkizCommandParam.HI: FAN_HIGH,
+        OverkizCommandParam.LOW: FAN_LOW,
+        OverkizCommandParam.LO: FAN_LOW,
+        OverkizCommandParam.MEDIUM: FAN_MEDIUM,  # fallback, state can be exposed as MEDIUM, new state = med
+        OverkizCommandParam.MED: FAN_MEDIUM,
+        OverkizCommandParam.SILENT: FAN_SILENT,
+    },
+    Protocol.HLRR_WIFI: {
+        OverkizCommandParam.AUTO: FAN_AUTO,
+        OverkizCommandParam.HIGH: FAN_HIGH,
+        OverkizCommandParam.LOW: FAN_LOW,
+        OverkizCommandParam.MEDIUM: FAN_MEDIUM,
+        OverkizCommandParam.SILENT: FAN_SILENT,
+    },
+}
+
+FAN_MODES_TO_OVERKIZ: dict[Protocol, dict[str, str]] = {
+    Protocol.OVP: {
+        FAN_AUTO: OverkizCommandParam.AUTO,
+        FAN_HIGH: OverkizCommandParam.HI,
+        FAN_LOW: OverkizCommandParam.LO,
+        FAN_MEDIUM: OverkizCommandParam.MED,
+        FAN_SILENT: OverkizCommandParam.SILENT,
+    },
+    Protocol.HLRR_WIFI: {
+        FAN_AUTO: OverkizCommandParam.AUTO,
+        FAN_HIGH: OverkizCommandParam.HIGH,
+        FAN_LOW: OverkizCommandParam.LOW,
+        FAN_MEDIUM: OverkizCommandParam.MEDIUM,
+        FAN_SILENT: OverkizCommandParam.SILENT,
+    },
+}
+
+# This device is more complex since there is an OVP and an HLLRWIFI version, with different state mappings.
+# We need to make all state strings lowercase, since Hi Kumo server returns capitalized strings for some states. (without a clear pattern)
+
+
+class HitachiAirToAirHeatPump(OverkizEntity, ClimateEntity):
+    """Representation of Hitachi Air To Air HeatPump."""
+
+    _attr_hvac_modes = [*HVAC_MODES_TO_OVERKIZ]
+    _attr_preset_modes = [PRESET_NONE, PRESET_HOLIDAY_MODE]
+    _attr_swing_modes = [*SWING_MODES_TO_OVERKIZ]
+    _attr_target_temperature_step = 1.0
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_translation_key = DOMAIN
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+
+        self._attr_supported_features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE
+            | ClimateEntityFeature.FAN_MODE
+            | ClimateEntityFeature.PRESET_MODE
+        )
+
+        self.protocol = cast(Protocol, self.device.protocol)
```

Comment:
Why do we need this?

Reply:
Hitachi has several air to air heat pump devices that use 2 protocol versions (OVP and HLLRWIFI) that have small differences in states mapping strings.

Reply:
Now hold on with the approval. I can only test 1 of the protocols (HLLRWIFI) with my device and we (@iMicknl) have found a tester for the other protocol (OVP) who is reporting a bug. We should fix it before merging.

Reply:
I am resolving this conversation as I have isolated both protocols and will make a separate PR for OVP